src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/ServiceProviderCommentFreeText.java

```diff
@@ -13,22 +13,22 @@
 @RequiredArgsConstructor
 @Getter
 @EqualsAndHashCode(callSuper = false)
-public class ServiceProviderComment extends Segment {
+public class ServiceProviderCommentFreeText extends Segment {
     private static final String KEY = "FTX";
     private static final String QUALIFIER = "SPC";
     private static final String KEY_QUALIFIER = KEY + PLUS_SEPARATOR + QUALIFIER;
     private static final int FREE_TEXT_INDEX = 4;
 
     private final String serviceProviderComment;
 
-    public static ServiceProviderComment fromString(String edifactString) {
-        if (!edifactString.startsWith(ServiceProviderComment.KEY_QUALIFIER)) {
-            throw new IllegalArgumentException("Can't create " + ServiceProviderComment.class.getSimpleName() + " from " + edifactString);
+    public static ServiceProviderCommentFreeText fromString(String edifactString) {
+        if (!edifactString.startsWith(ServiceProviderCommentFreeText.KEY_QUALIFIER)) {
```

Comment:
Minor note, the class qualifier is not necessary. Some style guides advice using it but we're not using any such rule.
```suggestion
        if (!edifactString.startsWith(KEY_QUALIFIER)) {
```

Reply:
Fair enough! I have removed it in other places as well where it has been used: 8485ff7df86496fa208d3b44f7529c4c5d1ebf46